contracts/src/rewards/RewardsManager.sol

```diff
@@ -0,0 +1,122 @@
+    RewardsAggregator public rewardsAggregator;
+    EpochManager public epochManager;
+    NodeRegistry public nodeRegistry;
+    SD59x18 public inflationInLastEpoch;
+
+    bool private initialized;
+
+    // epoch number => bool
+    mapping(uint256 => bool) public rewardsDistribution;
+
+    uint256 constant DAYS_IN_YEAR = 365;
+
+    // min factor by which the inflation can go down based on usage in %
+    SD59x18 public minInflationFactor;
+    // max inflation for the year in %
+    SD59x18 public maxInflation;
+    // price per byte
+    SD59x18 public price;
+    // Cost of running node
+    SD59x18 public cost;
+
+    event RewardMinted(address indexed account, uint256 amount);
+
+    function initialize(address _controller, address _token, address _epoch, address _aggregator, address _registry)
+        external
+    {
+        require(!initialized, "Rewards contract already initialized");
+        Controlled._init(_controller);
+        fleekToken = FleekToken(_token);
+        nodeRegistry = NodeRegistry(_registry);
+        epochManager = EpochManager(_epoch);
+        rewardsAggregator = RewardsAggregator(_aggregator);
+        inflationInLastEpoch = maxInflation;
+        initialized = true;
+    }
+
+    function setInflationRate(SD59x18 _inflationRate) external {
+        maxInflation = _inflationRate;
+    }
+
+    function setMinInflationFactor(SD59x18 _minInflationFactor) external {
+        minInflationFactor = _minInflationFactor;
+    }
+
+    function setPrice(SD59x18 _price) external {
+        price = _price;
+    }
+
+    function setCost(SD59x18 _cost) external {
+        cost = _cost;
+    }
+
+    /**
+     * @dev Distribute reward tokens to addresses.
+     * @param epoch epoch for which the rewards to be distributed
+     */
+    function distributeRewards(uint256 epoch) public onlyController {
+        require(epochManager.epoch() != epoch, "cannot distribute rewards for current epoch");
+        require(!rewardsDistribution[epoch], "rewards already distributed for this epoch");
+
+        SD59x18 _uActual = convert(int256(rewardsAggregator.getDataForEpoch(epoch)));
+        SD59x18 _uPotential = convert(int256(rewardsAggregator.getAvgUsageNEpochs(epoch)));
+
+        SD59x18 _totalMint = _getMintRate(_uActual, _uPotential);
+        // todo: variable distribution
+        // 75% goes to edge node
+        SD59x18 _toEdgeNode = _totalMint.mul(sd(0.75e18));
+        string[] memory publicKeys = rewardsAggregator.getPublicKeys(epoch);
+
+        for (uint256 i = 0; i < publicKeys.length; i++) {
+            uint256 dataServedByNode = rewardsAggregator.getDataServedByNode(publicKeys[i], epoch);
+            SD59x18 servedPercentage = convert(int256(dataServedByNode)).div(_uActual);
+            SD59x18 rewardsAmount = servedPercentage.mul(_toEdgeNode);
+            // check if the node with public key is white listed
+            (address _to,,,,) = nodeRegistry.whitelist(publicKeys[i]);
```

Comment:
```suggestion
            (address to,,,,) = nodeRegistry.whitelist(publicKeys[i]);
```
minor nit: nameing this var _to would imply that the variable was passed in as a paramater by solidity standards